File: js/marsExploration.js
/*
A space explorer's ship crashed on Mars! They send a series of SOS messages to Earth for help.

Letters in some of the SOS messages are altered by cosmic radiation during transmission. Given the signal received by Earth as a string, , determine how many letters of the SOS message have been changed by radiation.
*/

marsExploration = (s) => {
  let count = 0;
  let split = s.match(/.{1,3}/g);
  console.log('split', split);
  split.forEach((message) => {
    console.log('message', message);
    if(message !== 'SOS') {
      let splitMessage = message.split('');
      console.log('splitMessage', splitMessage);
      if(splitMessage[0] !== 'S') {
        count++;
        console.log('count', count);
      }
      if (splitMessage[1] !== 'O') {
        count++;
        console.log('count', count);
      }
      if (message[2] !== 'S') {
        count++;
        console.log('count', count);
      }
    }
  });
  console.log('return', count);
  return count;
}

console.log('testCase0', marsExploration('SOSSPSSQSSOR'));
console.log('testCase3', marsExploration('SOSOOSOSOSOSOSSOSOSOSOSOSOS'));

File: python/mars_exploration.py
'''
A space explorer's ship crashed on Mars! They send a series of SOS messages to Earth for help.

Letters in some of the SOS messages are altered by cosmic radiation during transmission. Given the signal received by Earth as a string, , determine how many letters of the SOS message have been changed by radiation.
'''

def mars_exploration(s):
  count = 0;
  n = 3;
  split = [s[i:i+n] for i in range(0, len(s), n)]
  print(split)
  for message in split:
    print(message)
    if message != 'SOS':
      split_message = [*message]
      print(split_message)
      if split_message[0] != 'S':
        count += 1
      if split_message[1] != 'O':
        count += 1
      if split_message[2] != 'S':
        count += 1
  print('return', count)
  return count

mars_exploration('SOSSPSSQSSOR')
mars_exploration('SOSOOSOSOSOSOSSOSOSOSOSOSOS')
